### PR TITLE
Fix scripts/check-{examples,workspace}

### LINKED ISSUE #455: Fix scripts/check-* and possibly rewrite them in Rust
The `scripts/check-*` files are intended to be run before pushing/creating a PR. This is also described [here](https://github.com/paritytech/ink#scripts) in the Readme. They currently do not work though, there are some errors and some things can be removed altogether since they've become outdated.

# ToDo List

- [ ] Fix `scripts/check-*`. For `check-workspace.sh` it seems the second clippy line is just missing wasm as a target.
- [ ] We removed the distinction between all and wasm crates a while ago. Adapt the scripts accordingly.
- [ ] Remove crates which no longer exist (cli, etc.).
- [ ] **Bonus:** Rewrite these check scripts in Rust using the [std::process crate](https://doc.rust-lang.org/std/process/index.htm).

### PR DESCRIPTION
**Pull request by cmichi**
Closes #455.

### CODE REVIEW

File: scripts/check-workspace.sh [hunk 1]
```diff
@@ -17,19 +17,17 @@
 
 declare -A results
 
-all_crates=("core" "alloc" "utils" "model" "lang" "lang" "lang/macro" "cli")
-wasm_crates=("core" "alloc" "utils" "model" "lang" "lang" "lang/macro")
+all_crates=("env" "storage" "storage/derive" "allocator" "prelude" "primitives" "lang" "lang/macro" "lang/ir")
```
Comment (Robbepop) [inline scripts/check-workspace.sh:20]: `lang/codegen` missing

Comment (Robbepop) [reply scripts/check-workspace.sh:20]: `metadata` missing

**Review by Robbepop — COMMENTED** [review]
All in all LGTM, some nits.

File: scripts/check-examples.sh [hunk 2]
```diff
@@ -37,24 +48,49 @@ for example in $(ls examples/lang); do
...
+metadata examples/delegator/
 
 banner="---------------"
 echo "Example Results"
 echo "$banner"
 for entry in ${!results_wasm[@]}; do
     echo "- $entry (wasm): ${results_wasm[$entry]}"
     echo "- $entry (test): ${results_test[$entry]}"
-    echo "- $entry (abi) : ${results_abi[$entry]}"
+    echo "- $entry (metadata) : ${results_metadata[$entry]}"
```
[the remark this answers is not in the document]
Comment (cmichi) [reply scripts/check-examples.sh:93]: Dang, why didn't my OCD notice this! Done!

File: scripts/check-workspace.sh (hunk 1 above)
Comment (cmichi) [reply scripts/check-workspace.sh]: So I copied this from out [.gitlab-ci.yml](https://github.com/paritytech/ink/blob/master/.gitlab-ci.yml#L20). Are they missing there intentionally or should we add them there as well?

File: scripts/check-examples.sh [hunk 3]
```diff
@@ -37,24 +48,49 @@ for example in $(ls examples/lang); do
...
 banner="---------------"
 echo "Example Results"
 echo "$banner"
 for entry in ${!results_wasm[@]}; do
-    echo "- $entry (wasm): ${results_wasm[$entry]}"
-    echo "- $entry (test): ${results_test[$entry]}"
-    echo "- $entry (abi) : ${results_abi[$entry]}"
+    echo "- $entry (wasm):      ${results_wasm[$entry]}"
+    echo "- $entry (test):      ${results_test[$entry]}"
+    echo "- $entry (metadata) : ${results_metadata[$entry]}"
```
Comment (Robbepop) [inline scripts/check-examples.sh:93]:
```suggestion
    echo "- $entry (wasm):     ${results_wasm[$entry]}"
    echo "- $entry (test):     ${results_test[$entry]}"
    echo "- $entry (metadata): ${results_metadata[$entry]}"
```

**Review by Robbepop — COMMENTED** [review]
one tiny formatting nit left, then we can merge

**Review by Robbepop — APPROVED** [review]

**Pull request merged**